varianteval/core/io/vcf.py

```
@@ -53,9 +61,136 @@ def vcf2callset(vcf_fname, filter_funcs=None):
+        else:
+            return BREAKPOINT_SIDE.RIGHT
+
+
+def get_confidence_interval(record: VariantRecord, is_first_pos: bool, sigma_multiple: int) -> tuple[int, int]:
+    """
+    Returns the range of uncertainty around a position ``X`` (implicit from the 
+    context).
+    
+    :param is_first_pos: TRUE=first position in a VCF record; FALSE=last 
+    position;
+    :param sigma_multiple: if the record does not contain coordinates but just 
+    the STD, the procedure assumes an interval that extends this multiple of the
+    STD on both sides (2 or 3 captures most of a Gaussian);
+    :return: First element: quantity to be added to ``X`` to get the first 
+    position of the uncertainty interval (typically negative or zero). Second 
+    element: quantity to be added to ``X`` to get the last position of the 
+    uncertainty interval (typically positive or zero).
+    """
+    quantum: int
+    value: float	
+
+    if is_first_pos:
+        if VCF_CIPOS_STR in record.info.keys():
+            return tuple(map(int, record.info.get(VCF_CIPOS_STR).split(VCF_CI_SEPARATOR)))
+            # The sign is already ok
+        for string in VCF_STD_START:
+            if string in record.info.keys():
+                value = float(record.info.get(string))
+                quantum = int(value*SIGMA_MULTIPLE)
+                return (-quantum, quantum)
+    else:
+        if VCF_CIEND_STR in record.info.keys():
+            return tuple(map(int, record.info.get(VCF_CIEND_STR).split(VCF_CI_SEPARATOR)))
+            # The sign is already ok
+        if VCF_CILEN_STR in record.info.keys():
+            return tuple(map(int, record.info.get(VCF_CILEN_STR).split(VCF_CI_SEPARATOR)))
+            # The sign is already ok
+        for string in VCF_STD_END:
+            if string in record.info.keys():
+                value = float(record.info.get(string))
+                quantum = int(value*SIGMA_MULTIPLE)
+                return (-quantum, quantum)
+    return (0, 0)
```

nit: when do we expect to reach this? if expected, can document in the function's return statement; if not, should we throw an error?

---

It is reached when the record contains no uncertainty information (so we tacitly assume zero uncertainty). The :return: documentation already says that any element of the returned tuple can be zero.

---

can we add when it will be zero in the docs

---

Done